pypesto/objective/priors.py

```diff
@@ -0,0 +1,264 @@
+import scipy.stats
+import numpy as np
+from typing import Optional, Union, Iterable, Dict, Callable, List, Tuple
+from copy import deepcopy
+import math
+
+from .function import Objective
+from .aggregated import AggregatedObjective
+
+
+class Priors(AggregatedObjective):
+    """
+    Handles prior distributions.
+
+    Consists basically of a list of individual priors,
+    given in self.objectives.
+    """
+
+    def __init__(self,
+                 prior_list: List[Objective],
+                 x_names: List[str] = None):
+
+        super().__init__(prior_list, x_names)
+
+
+class ParameterPriors(Objective):
+    """
+    Single Parameter Prior.
+
+
+    prior_list has to contain dicts of the format format
+    {'index': [int], 'density_fun': [Callable], 'density_dx': [Callable], 'density_ddx': [Callable]}
+
+
+    Note:
+    -----
+    All callables should correspond to (log)densities and are internally
+    multiplied by -1, since pyPESTO performs minimization...
+    """
+
+    def __init__(self,
+                 prior_list: list):
+
+        self.prior_list = prior_list
+
+        super().__init__(fun=self.density_for_full_parameter_vector,
```

Comment:
I think this reference to `self` for things that are passed to the super method is unusual and potentially leads to errors when rebinding, similarly to what we had in AmiciObjective until the recent changes (@FFroehlich?) It would be better to not refer to own methods here. If ParameterPriors is re-implemented to only host a single prior (see above), this would be straightforward to do.

Reply:
yes, these functions will bind to subsequent updates to the prior. Would be addressed by using `ObjectiveBase` as base class and  implementing `call_unprocessed`. I am also wondering whether we could implement "residual" versions of priors such that `MODE_RES` would be supported. For most basic priors this should be straightforward. Fine with me if this is addressed at a later point though.